packages/desktop-client/src/components/modals/EditRuleModal.jsx

```diff
@@ -651,7 +653,7 @@ function ConditionsList({
   ) : (
     <Stack spacing={2} data-testid="condition-list">
       {conditions.map((cond, i) => {
-        let ops = TYPE_INFO[cond.type].ops;
+        let ops = getValidOps(cond.field);
```

Comment:
This could be done in a separate refactor, but right now there is no way for `getValidOps` to perform op-specific logic. The change would also need additional testing that I don't want to add to an already-large PR. But thanks for the suggestion! I made a note of it.